protos/client.proto

```diff
@@ -50,72 +50,80 @@ message ClientStateCurrentResponse {
     }
     Status status = 1;
     string merkle_root = 2;
-
 }
 
-// A request from a client for a particular address's data, for a given merkle root
-message ClientStateGetRequest {
+// A request to list every entry under a given merkle root or head block.
+message ClientStateListRequest {
     string merkle_root = 1;
-    string address = 2;
+    string subtree = 2;
 }
 
-// The response to a GetRequest from the client. NORESOURCE means that either the
-// merkle root is not a merkle root in State or the address doesn't exist
-// under a valid merkle root. NONLEAF means that the address isn't a valid
-// leaf address in the merkle tree, this would probably mean that the address
-// is a prefix (truncated). ERROR is a general internal error, like the protobuf
-// sent by the client didn't deserialize correctly.
-message ClientStateGetResponse {
+// A response that lists the Entries under a given merkle root.
+//
+// Statuses:
+//   * OK - everything worked as expected
+//   * ERROR - general internal error, such as protobuf failing to deserialize
+//   * NOGENESIS - the validate has not been configrued with a genesis block
+//   * NOROOT - the merkle_root specified was not found
+//   * NORESOURCE - the root sprecidied is valid, but contains no data
+
+message ClientStateListResponse {
     enum Status {
         OK = 0;
-        NORESOURCE = 1;
-        NONLEAF = 2;
-        ERROR = 3;
+        ERROR = 1;
+        NOROOT = 3;
+        NORESOURCE = 2;
+        NOGENESIS = 4;
```

Comment:
Yes. The client reports back that the validator has no genesis block yet. Given that it is possible to distinguish between this and a general error, I erred on the side of being more informative.